packages/beacon-node/src/chain/blocks/verifyBlock.ts

```diff
@@ -38,249 +31,79 @@ export type VerifyBlockModules = {
-
-        const parentRoot = toHexString(block.message.parentRoot);
-        const parentBlock = chain.forkChoice.getBlockHex(parentRoot);
-        const justifiedBlock = chain.forkChoice.getJustifiedBlock();
+  // TODO: Skip in process chain segment
+  // Retrieve preState from cache (regen)
+  const preState0 = await chain.regen.getPreState(block0.message, RegenCaller.processBlocksInEpoch).catch((e) => {
+    throw new BlockError(block0, {code: BlockErrorCode.PRESTATE_MISSING, error: e as Error});
+  });
 
-        if (
-          !parentBlock ||
-          // Following condition is the !(Not) of the safe import condition
-          (parentBlock.executionStatus === ExecutionStatus.PreMerge &&
-            justifiedBlock.executionStatus === ExecutionStatus.PreMerge &&
-            block.message.slot + opts.safeSlotsToImportOptimistically > chain.clock.currentSlot)
-        ) {
-          throw new BlockError(block, {
-            code: BlockErrorCode.EXECUTION_ENGINE_ERROR,
-            execStatus: ExecutePayloadStatus.UNSAFE_OPTIMISTIC_STATUS,
-            errorMessage: `not safe to import ${execResult.status} payload within ${opts.safeSlotsToImportOptimistically} of currentSlot, status=${execResult.status}`,
-          });
-        }
+  // Ensure the state is in the same epoch as block0
+  if (block0Epoch !== computeEpochAtSlot(preState0.slot)) {
+    throw Error(`preState at slot ${preState0.slot} must be dialed to block epoch ${block0Epoch}`);
+  }
 
-        executionStatus = ExecutionStatus.Syncing;
-        break;
-      }
+  const abortController = new AbortController();
 
-      // If the block has is not valid, or it referenced an invalid terminal block then the
-      // block is invalid, however it has no bearing on any forkChoice cleanup
-      //
-      // There can be other reasons for which EL failed some of the observed ones are
-      // 1. Connection refused / can't connect to EL port
-      // 2. EL Internal Error
-      // 3. Geth sometimes gives invalid merkel root error which means invalid
-      //    but expects it to be handled in CL as of now. But we should log as warning
-      //    and give it as optimistic treatment and expect any other non-geth CL<>EL
-      //    combination to reject the invalid block and propose a block.
-      //    On kintsugi devnet, this has been observed to cause contiguous proposal failures
-      //    as the network is geth dominated, till a non geth node proposes and moves network
-      //    forward
-      // For network/unreachable errors, an optimization can be added to replay these blocks
-      // back. But for now, lets assume other mechanisms like unknown parent block of a future
-      // child block will cause it to replay
+  try {
+    const [{postStates, proposerBalanceDeltas}, , {executionStatuses, mergeBlockFound}] = await Promise.all([
+      // Run state transition only
+      // TODO: Ensure it yields to allow flushing to workers and engine API
```

Comment:
remove this `TODO`?

Reply:
This goal is not accomplished, but it's just a hope. I haven't found a way to ensure from JS land that data has been sent to the worker which is the ultimate issue that cap throughput.